llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -15694,6 +15703,9 @@ static SDValue expandMul(SDNode *N, SelectionDAG &DAG,
     }
   }
 
+  if (SDValue V = expandMulToAddOrSubOfShl(N, DAG, MulAmt))
```

Comment:
and is there any reason you moved this below strength reduction via division? (line 15681 ~ 15704)

Reply:
I vaguely remember there being a reason I broke the conditions in this order, but it looks like there's no test diffs resulting from this, so it's probably(?) fine.